src/sentry/tasks/assemble.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import, print_function
+
+import six
+import logging
+
+from sentry.tasks.base import instrumented_task
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(name='sentry.tasks.assemble.assemble_chunks', queue='assemble')
+def assemble_chunks(type, params, file_id, file_blob_ids, checksum, **kwargs):
+    '''This assembles multiple chunks into on File.
+    The type is a File.ChunkAssembleType
+    '''
+    if len(file_blob_ids) == 0:
+        logger.warning('sentry.tasks.assemble.assemble_chunks', extra={
+            'error': 'Empty file blobs'
+        })
+
+    from sentry.models import File
+    from sentry.models.file import ChunkFileState, ChunkAssembleType
+    file = File.objects.filter(
+        id=file_id,
+    ).get()
+
+    file.headers['state'] = ChunkFileState.ASSEMBLING
+    # Do the actual assembling here
+    file.assemble_from_file_blob_ids(file_blob_ids, checksum)
+    if file.headers.get('state', '') == ChunkFileState.ERROR:
+        logger.error(
+            'sentry.tasks.assemble.assemble_chunks',
```

Comment:
Since you have multiple `assemble_chunks` error statements throughout your code, you should append them with the logical reason they're erroring, so `assemble_chunks.state_error` or something.